DESIGN.md

```diff
@@ -1,4 +1,73 @@
 # PowerShell Manager: Design and Architecture
 
-<!-- TODO: Fill this out! -->
-<!-- NOTE: Not multi-thread safe, executions against the same manager in parallel can get _weird_ -->
\ No newline at end of file
+<!-- NOTE: Not multi-thread safe, executions against the same manager in parallel can get _weird_ -->
+
+Managing Windows infrastructure from ruby is difficult - much of the interaction with the tools and APIs is performed via PowerShell, and PowerShell remains the most common language for operators of Windows machines to know.
+Calling PowerShell from ruby has a few problems, but the most painful one is speed - shelling out requires spinning up the PowerShell process, executing a single pipeline, and tearing it back down.
+
+This gem allows the use of a long-lived manager to which ruby can send PowerShell invocations and receive the exection output.
+This reduces the time to execute a second command from seconds to milliseconds.
```

Comment:
` execute a second command from seconds to milliseconds.`

Perhaps reword.  second vs seconds was initially confusing.  Perhaps "subsequent" or "next"

Reply:
To further clarify, we should add something along the lines of 'this reduces the time to execute *the* command, not the total time it takes the command to execute.'

Reply:
Resolved with a substantial reword, I think.